Add release module

theory opened the pull request:
The new `Release` struct encapsulates `Distribution` and adds a `release` field that contains JWS-signed release metadata. It currently only supports v2 metadata, but offers the same traits, constructors, and accessors as `Distribution`, plus the `release()` accessor and associated objects.

Add tests for all of these bits, using both the v2 corpus with a release patch applied and custom JSON objects to test each individual struct and its accessors.

Writing the tests for the `TryFrom<PathBuf>` when what was on hand was a `Path` led to a bit of research and the conclusion that one does not convert a file path into a struct, but loads it into a struct. So create the `load` function, instead, and have it accept a `AsRef<Path>` argument, which covers `Path`s, `PathBuf`s, and strings. Back-patch this change to `Distribution`, as well, and take advantage of `.try_into()` where possible.

vrmiguel commented on src/release/mod.rs:
```diff
@@ -0,0 +1,472 @@
+
+impl Digests {
+    /// Borrows the SHA-1 hash.
+    pub fn sha1(&self) -> Option<&[u8]> {
```
Another option:

```suggestion
    pub fn sha1(&self) -> Option<&[u8; 20]> {
        self.sha1.as_ref()
    }
```

theory replied:
Ahhh, thank you! Switched to that in 88b2183.

theory committed:
Add release module

The new `Release` struct encapsulates `Distribution` and adds a
`release` field that contains JWS-signed release metadata. It currently
only supports v2 metadata, but offers the same traits, constructors, and
accessors as `Distribution`, plus the `release()` accessor and
associated objects.

Add tests for all of these bits, using both the v2 corpus with a release
patch applied and custom JSON objects to test each individual struct and
its accessors.

Writing the tests for the `TryFrom<PathBuf>` when what was on hand was a
`Path` led to a bit of research and the conclusion that one does not
convert a file path into a struct, but loads it into a struct. So create
the `load` function, instead, and have it accept a `AsRef<Path>`
argument, which covers `Path`s, `PathBuf`s, and strings. Back-patch this
change to `Distribution`, as well, and take advantage of `.try_into()`
where possible.
